Dialog/LevelConfigDialog.cpp

```diff
@@ -1,11 +1,63 @@
-#include "LevelConfigDialog.h"
+﻿#include "LevelConfigDialog.h"
 #include "ui_LevelConfigDialog.h"
 
-#include <QPushButton>
-
-QRegExp LevelnameRegx("^[A-Za-z0-9\\s]+$");
+#include "LevelComponents/Level.h"
 QRegExp TimerRegx("^[0-9]:[0-5][0-9]$");
 
+/// <summary>
+/// Enable or disable the OK button depending on the validity of the input fields.
+/// All fields must be in the QValidator::State::Acceptable, or else the OK button will be disabled.
+/// </summary>
+/// <param name="textBoxes">
+/// The text boxes to validate.
+/// </param>
+/// <param name="okButton">
+/// The OK button which will be enabled or disabled based on the text boxes' validator states.
+/// </param>
+void LevelConfigDialog::SetOKButtonEnable(QVector<QLineEdit *> textBoxes, QPushButton *okButton)
+{
+    bool allValid = true;
+    for (auto iter = textBoxes.begin(); iter != textBoxes.end(); ++iter)
+    {
+        QLineEdit *line = *iter;
+        auto variable = line->text();
+        int pos = 0;
+        if (line->validator()->validate(variable, pos) != QValidator::State::Acceptable)
+        {
+            allValid = false;
+            break;
+        }
+    }
+    if(!allValid)
+    {
+        okButton->setEnabled(allValid);
+        return;
+    }
+    QString charstable = LevelComponents::Level::GetAvailableLevelNameChars();
+    for (int k = 0; k < 2; k++)
+    {
+        QString LevelName = k ? ui->LevelNameJ_TextBox->text() : ui->LevelName_TextBox->text();
+        for (int i = 0; i < LevelName.size(); i++)
+        {
+            for (int j = 0; j < charstable.size(); j++)
```

Comment:
Let's remove this for loop and do `charstable.contains(LevelName[i])`

https://doc.qt.io/qt-5/qstring.html#contains-1

Reply:
Also, the space character logic should be a part of charstable

Reply:
no, there a several blank slots, not only one.